packages/schema/lib/functional-constraints/labelWhenVisible.js

```diff
@@ -0,0 +1,31 @@
+const _ = require('lodash');
+const jsonschema = require('jsonschema');
+
+const actionTypes = ['triggers', 'searches', 'creates'];
+
+const labelWhenVisible = definition => {
+  const errors = [];
+
+  for (const actionType of actionTypes) {
+    const group = definition[actionType] || {};
+    _.each(group, (action, key) => {
+      const { display } = action;
+      if (!display.hidden && !(display.label && display.description)) {
+        errors.push(
+          new jsonschema.ValidationError(
+            `visible actions must have a label and description`,
+            action,
+            `/${_.capitalize(actionType)}Schema`,
+            `instance.${key}.key`,
```

Comment:
Ideally, I think the path should be `instance.${key}.display.label` or `instance.${key}.display.description` to be more accurate, but I'm ok with `instance.${key}.display` too.

Reply:
went with `instance.creates.foo.display`, which is the path to where the dev will need to fix the issue. Good catch!